pkgs/development/python-modules/detectron2/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonRelaxDepsHook
+, ninja
+# build inputs
+, pillow
+, matplotlib
+, pycocotools
+, termcolor
+, yacs
+, tabulate
+, cloudpickle
+, tqdm
+, tensorboard
+, fvcore
+, iopath
+, omegaconf
+, hydra-core
+, black
+, packaging
+# optional dependencies
+, fairscale
+, timm
+, scipy
+, shapely
+, pygments
+, psutil
+}:
+
+let
+  name = "detectron2";
+  version = "0.6";
+  optional-dependencies = {
+    all = [
+      fairscale
+      timm
+      scipy
+      shapely
+      pygments
+      psutil
+    ];
+  };
+in
+buildPythonPackage {
+  inherit name version;
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "1w6cgvc8r2lwr72yxicls650jr46nriv1csivp2va9k1km8jx2sf";
+  };
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+    ninja
+  ];
+
+  dontUseNinjaBuild = true;
+  dontUseNinjaInstall = true;
+  dontUseNinjaCheck = true;
+
+  pythonRelaxDeps = [
+    "black"
+  ];
+
+  propagatedBuildInputs = [
+    pillow
+    matplotlib
+    pycocotools
+    termcolor
+    yacs
+    tabulate
+    cloudpickle
+    tqdm
+    tensorboard
+    fvcore
+    iopath
+    omegaconf
+    hydra-core
+    black
```

Comment:
Do we really need a formatter at runtime?

Reply:
That's what they seem to think about
```
 ERROR: Could not find a version that satisfies the requirement black (from detectron2) (from versions: none)
       > ERROR: No matching distribution found for black
```
I didn't dig on this one, we could relax the dependency. let me have a look.

Reply:
It is actually required at runtime https://github.com/facebookresearch/detectron2/blob/47d8ed7a1bd999719825f2e8a11d5b2828bc02c4/detectron2/config/lazy.py#L433C1-L436